Framework/LiveData/src/Kafka/KafkaEventListener.cpp

```diff
@@ -13,21 +14,32 @@ namespace LiveData {
 
 DECLARE_LISTENER(KafkaEventListener)
 
-KafkaEventListener::KafkaEventListener() {
-  declareProperty("InstrumentName", "");
+void KafkaEventListener::setAlgorithm(
+    const Mantid::API::IAlgorithm &callingAlgorithm) {
+  this->updatePropertyValues(callingAlgorithm);
+  // Get the instrument name from StartLiveData so we can sub to correct topics
+  if (callingAlgorithm.existsProperty("Instrument")) {
+    m_instrumentName = callingAlgorithm.getPropertyValue("Instrument");
+  } else {
+    g_log.error("KafkaEventListener requires Instrument property to be set in "
+                "calling algorithm");
+  }
 }
 
 /// @copydoc ILiveListener::connect
 bool KafkaEventListener::connect(const Poco::Net::SocketAddress &address) {
+  if (m_instrumentName.empty()) {
```

Comment:
What happens if the instrument name doesn't match a topic?

Reply:
This is checked by the consumer, an error is logged and the algorithm terminates.